Tests/Unit/Encodings/TransferPayloadTests.swift

```diff
@@ -0,0 +1,191 @@
+//
+//  Copyright (c) 2020-2021 MobileCoin. All rights reserved.
+//
+
+import LibMobileCoin
+@testable import MobileCoin
+import XCTest
+
+class TransferPayloadTests: XCTestCase {
+
+    // The base64Encoded string below will decode to 32 bytes of data for a valid Data32
+    static var entropyData = Data(base64Encoded: "ajaEQTHHDeZEZDk1rGYQRF0ErcpmcPa7buRpNchz4hQ=")!
```

Comment:
we should avoid force-casts when possible.

Reply:
Ya, this was annoying me a bit too....

I didn't want to put these (e.g. also the ristrettoCompressed) values into the local functions - not sure how costly, say, the External_Compressed(..) call is...

Reply:
The existing pattern is to define these things in a "Fixtures" file, then reference them in the tests:

```
  Tests/Common/Fixtures/Network/Connection/TestBlockchainService.swift
  Tests/Common/Fixtures/Network/Attestation/Attestation+Fixtures.swift
  Tests/Common/Fixtures/Network/Attestation/AttestAke+Fixtures.swift
  Tests/Common/Fixtures/Crypto/VersionedCryptoBox+Fixtures.swift
  Tests/Common/Fixtures/Transaction/Transaction+Fixtures.swift
  Tests/Http/FixturesData/ConnectionSession+HttpFixtures.swift
  Tests/Grpc/FixturesData/ConnectionSession+GrpcFixtures.swift
  Tests/Common/Fixtures/Fog/Report/FogResolver+Fixtures.swift
  Tests/Common/Fixtures/Account/PublicAddress+Fixtures.swift
  Tests/Common/Fixtures/Network/NetworkConfig+Fixtures.swift
  Tests/Common/Fixtures/Transaction/Receipt+Fixtures.swift
  Tests/Common/Fixtures/Fog/View/FogRngKey+Fixtures.swift
  Tests/Common/Fixtures/Account/AccountKey+Fixtures.swift
  Tests/Common/Fixtures/MobileCoinClient+Fixtures.swift
  Tests/Common/Fixtures/Fog/View/FogRng+Fixtures.swift
  Tests/Common/Fixtures/Account/Account+Fixtures.swift
  Tests/Common/Fixtures/Network/NetworkPreset.swift
  Tests/Common/Fixtures/Ledger/TxOut+Fixtures.swift
  Tests/Common/Fixtures/Encodings/Base58Coder.swift
  Tests/Common/Fixtures/Encodings/MobUri.swift
```

Its a decent way to get all the static test data into one place so it can be re-used.

Reply:
Awesome - switched to using Fixtures for the test data.

Also, if the test data creation fails, I throw an exception with specific context around what, exactly, failed, and this info will then be logged to the test output so it will be easy to debug.

e.g., a dev accidentally deletes a character from the base64 encoded strings used to generate a test value, so the test value is nil....

I didn't want to add checks to also verify the test data within the tests... but the values *could* be nil, and that may still allow the tests to pass (b/c the verified values will still be equal). 

This way, if something changes or is wrong with the input, the tests will fail, and the specific problem will be reported to save time debugging.